nvflare/private/fed/server/job_runner.py

```diff
@@ -338,10 +338,16 @@ def run(self, fl_ctx: FLContext):
                                         ready_job.job_id, {JobMetaKey.JOB_DEPLOY_DETAIL.value: deploy_detail}, fl_ctx
                                     )
 
+                                if failed_clients:
+                                    deployable_clients = {k: v for k, v in client_sites.items()
+                                                          if k not in failed_clients}
+                                else:
+                                    deployable_clients = client_sites
+
                                 self._start_run(
                                     job_id=job_id,
                                     job=ready_job,
-                                    client_sites=client_sites,
+                                    client_sites=deployable_clients,
                                     fl_ctx=fl_ctx,
                                 )
```

Comment:
Maybe we need to check if a job is deployable at scheduler time.

Cause we need to also check if require_clients is in deployable_clients
and if deployable_clients are larger than min_clients

Reply:
got it, make sense to me, agree to keep this for now.